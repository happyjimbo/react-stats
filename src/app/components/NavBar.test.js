import React from 'react'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'
import Navbar from './Navbar'

describe('NavBar', () =>{

    it('compares component to snapshot', () => {        
        const component = renderer.create(<Navbar />)
        const navBar = component.toJSON()
        expect(navBar).toMatchSnapshot()
    })

	it('contains the Navbar className', () => {		
		const component = shallow(<Navbar />)

		const app = component.find(div => div.hasClass('NavBar'))
		expect(app).toBeTruthy()		
    })

    it ('run callback when Button clicked, setting value to true', () => {

        let clicked = false
        const selectClick = (textInput) => { clicked = true; }

        const props = { selectClick }
        const component = shallow(<Navbar {...props} />)


        expect(clicked).toBe(false)

        const button = component.find('Button')
        button.simulate('click')

        expect(clicked).toBe(true)
    })
})
