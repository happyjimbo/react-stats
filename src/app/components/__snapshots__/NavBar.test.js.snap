exports[`NavBar compares component to snapshot 1`] = `
<div
  className="NavBar">
  <nav
    className="navbar navbar-default">
    <div
      className="container">
      <div
        className="navbar-header">
        <span
          className="navbar-brand">
          Stocks App
        </span>
        <button
          className="navbar-toggle collapsed"
          onClick={[Function]}
          type="button">
          <span
            className="sr-only">
            Toggle navigation
          </span>
          <span
            className="icon-bar" />
          <span
            className="icon-bar" />
          <span
            className="icon-bar" />
        </button>
      </div>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse">
        <div
          className="navbar-form navbar-left">
          <div
            className="form-group"
            onChange={[Function]}>
            <input
              className="form-control"
              id={undefined}
              placeholder="stock"
              type="text" />
          </div>
           
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="submit">
            Search
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
